doc/workflows.rst

```diff
@@ -0,0 +1,505 @@
+the change is considered non-significant can be modified using ``--alpha``,
+which will set the alpha risk of the Fisher's exact test carried out on a
+contingency table like this one:
+
+.. table:: Contingency table for one testcase
+  :widths: auto
+
+  ======   =====  =====
+  count     old    new
+  ======   =====  =====
+  passed   15     80
+  failed   5      20
+  ======   =====  =====
+
+That would represent an ``old`` test session with 20 iterations of the test, 15
+of which passed and 5 failed. The ``new`` session would have had 100
+iterations, out of which 80 passed and 20 failed.
+
+.. seealso:: :class:`lisa.regression.RegressionResult`
+
+The output of ``exekall compare`` looks like that:
+
+::
+
+    testcase                                                           old%   new% delta%      pvalue
+    -------------------------------------------------------------------------------------------------
+    PELTTask:test_load_avg_behaviour                                   3.8%   6.7%   2.9%    9.26e-02
+    ThreeSmallTasks:test_task_placement                                0.0%   1.8%   1.8%    4.13e-03
+    TwoBigTasks:test_slack                                             0.0%   3.1%   3.1%    1.03e-04
+    TwoBigThreeSmall:test_slack                                       82.1%  95.7%  13.6%    1.69e-06
+    TwoBigThreeSmall:test_task_placement                              79.7%  95.7%  16.0%    7.18e-08 
+
+The columns have the following meaning:
+
+  * ``old%``: failure rate of the test in the old database (i.e. the first on the command line)
+  * ``new%``: failure rate of the test in the new database (i.e. the second on the command line)
+  * ``delta%``: the difference in the old and new failure rates
+  * ``pvalue``: The p-value resulting from the Fisher's exact test used to
+    filter significant regressions or improvements
+
+.. tip:: When comparing results collected from different boards, the test IDs
+  will probably not match since they are tagged with the user-defined board
+  name. In order to overcome that, use ``--remove-tag board``, so IDs can be
+  matched as expected.
+
+Advanced use
+++++++++++++
+
+Parametric sweep
+----------------
+
+``--sweep`` allows running the same stage multiple times, sweeping over a range
+of values for some of its parameters:
+
+.. code-block:: sh
+
+  # The energy_est_threshold_pct parameter of functions with a name matching 
+  # '*test_task_placement' will take the following values all values from 0 to 15
+  # by increments of 5.
+  exekall run lisa/tests/ --conf target_conf.yml --sweep '*test_task_placement' energy_est_threshold_pct 0 15 5
+
+When something went wrong
+-------------------------
+
+``--replay`` provides a simple way of re-executed the last few stages of a test
+that had an error. That can be used to reproduce a bug in the test code that
+makes it raise an exception while working on a fix. ``--replay`` takes the UUID
```

Comment:
I'd add something to point out that this will not re-run the workload:
```suggestion
makes it raise an exception while working on a fix without having to re-execute the workload on a board (and thus without needing a board at all). ``--replay`` takes the UUID
```

Reply:
That is not necessarily true. If the exception happened when parsing the target configuration or platform info, it will restart from there and will use the board. It depends where the failure happened, the only guarantee you have is that all that can be "salvaged" will be.

You could even come up with trickier cases where the last stage wants a reference to the TestEnv. Since that cannot be serialized, it would have to be rebuilt, even if "intermediate" stages  like the TestBundle are available .

Reply:
True, I wanted to add that bit as an example rather than a forever true statement (the sentence does start with `that *can* be used to`)

Reply:
got it